app/gui/view/graph-editor/src/new_node_position.rs

```diff
@@ -113,7 +127,42 @@ pub fn below_line_and_left_aligned(
     let y_offset = y_gap + node::HEIGHT / 2.0;
     let starting_point = Vector2(align_x, line_y - y_offset);
     let direction = Vector2(-1.0, 0.0);
-    on_ray(graph_editor, starting_point, direction).unwrap()
+    let pos = on_ray(graph_editor, starting_point, direction).unwrap();
+    // Perform Magnet Alignment if the node is pushed left to the free space.
+    if pos.x != align_x {
```

Comment:
instead of comments, do variables. `was_pushed_left = ...`

Reply:
Done!